### ISSUE TITLE
Copy-Paste broken into VSCode Integrated Terminal 

### ISSUE THREAD

**Issue by maxruby** [Microsoft/vscode#18967]
- VSCode Insiders Version: 1.9.0
- OS Version: OSX 10.10.2

Steps to Reproduce:

1. Copy & Paste text from the VS Code Editor window into the Integrated Terminal
 -> Nothing appears to be pasted!
2.  Press `Enter` -> the next line shows the text that was pasted before

This problem came up only after updating VSCode Insiders to the latest version yesterday. I am not using any special `tmux` session.

**Comment by Tyriar** [comment]
Likely a duplicate of https://github.com/Microsoft/vscode/issues/18833

@maxruby I'm having difficulty reproducing. Can you post a screenshot of before/after steps 1 and 2? Also could you check dev tools console for errors/warnings? (Help > Toggle Developer Tools)

**Comment by maxruby** [comment]
@Tyriar.  
1.  Screenshot of the Editor with the Integrated Terminal after `copy and pasting` target text (red rectangle)
<img width="1440" alt="before" src="https://cloud.githubusercontent.com/assets/5504973/22183471/81ce082e-e0bf-11e6-9cbe-1fb849e1fd0a.png">
2. Screenshot after pressing `Enter`. Notice the empty line in the Integrated Terminal and the text now displayed in the second line.
<img width="1440" alt="after_copy_paste_into_terminal" src="https://cloud.githubusercontent.com/assets/5504973/22183492/f5f60328-e0bf-11e6-8ed8-5b3967413ad3.png">

**Comment by maxruby** [comment]
Sorry, accidentally closed the issue. 

Regarding debugging and the Developer Tools Console, there are no warnings or errors. 
However, I had a quick go at debugging myself in an attempt to understand what might be going on.  Its obvious that there is no text pasted into new `<span>` elements within the `<div>` that makes up each rows in the terminal, by querying in the `Console`:

```
var term = document.getElementsByClassName('xterm-rows');

[<div class=​"xterm-rows" style=​"line-height:​ 13.8889px;​">
​<div>​
<span class=​"xterm-bold xterm-color-10">​➜&nbsp;&nbsp;​</span>
​<span class=​"xterm-bold xterm-color-14">​~​</span>​"&nbsp;"
<span class=​"reverse-video terminal-cursor">​&nbsp;​</span>​
"&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"</div>

<span class=​"reverse-video terminal-cursor">​&nbsp;​</span>​
"&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"</div>
```

Applying a Breakpoint in `Clipboard: paste` under `Event Listener Breakpoints` stops in `xterm.js`, `line 488`:
```
on(this.textarea, 'paste', function (ev) {
    pasteHandler.call(this, ev, term);
  });
```
returns the following ***after***  Stepping Over the breakpoint in `line 489`: 
```
ev = ClipboardEvent {isTrusted: true, clipboardData: DataTransfer, type: "paste", target: textarea.xterm-helper-textarea, currentTarget: textarea.xterm-helper-textarea…}
```
Sorry, but I do not have time to further track this down now.

**Comment by Tyriar** [comment]
Thanks @maxruby, the 3 people who have experienced this all seem to be using zsh so that's likely the cause. Closing as duplicate of https://github.com/Microsoft/vscode/issues/18833

**Comment by Tyriar** [comment]
@maxruby do you use regular zsh or oh-my-zsh?

**Comment by maxruby** [comment]
Using `oh-my-zsh`.

**Comment by Tyriar** [comment]
Thanks
